src/w3d/renderer/render2dsentence.cpp

```diff
@@ -831,7 +836,10 @@ void Render2DSentenceClass::Build_Sentence_Centered(const unichar_t *text, int *
             sentence_width += word_width;
         }
 
-        m_cursor.X = std::max((position.X - sentence_width) * 0.5f, 0.0f);
+        m_cursor.X = (float)(int)((position.X - sentence_width) * 0.5f);
+        if (m_cursor.X < 0.0f) {
+            m_cursor.X = 0.0f;
+        }
```

Comment:
This change fixes the screwed character positions.

Reply:
You could still use max but make it cast to int to force the truncation

Reply:
Yeah, but this compares better with the decompile, so can leave as is. Important is that it works correctly.

Reply:
It is much harder to read and reason with when you add unnecessary casts at least drop the float cast there is no need to do that twice.

Reply:
Ok. Changed and tested.